src/st/no0/e_slinger.c

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "no0.h"
+
+typedef enum {
+    SLINGER_INIT,
+    SLINGER_IDLE,
+    SLINGER_WALK_TOWARDS_PLAYER,
+    SLINGER_WALK_AWAY_FROM_PLAYER,
+    SLINGER_ATTACK,
+    SLINGER_JUMP,
+    SLINGER_DESTROY
+} SLINGER_STEPS;
+
+typedef enum {
+    SLINGER_JUMPING,
+    SLINGER_IN_AIR,
+    SLINGER_LAND
+} SlingerJumpSubSteps;
+
+static u8 anim_walk[] = {0x06, 0x01, 0x04, 0x02, 0x04, 0x03, 0x06,
+                         0x04, 0x05, 0x05, 0x05, 0x06, 0x00};
+static u8 anim_walk_backwards[] = {0x06, 0x01, 0x05, 0x06, 0x05, 0x05, 0x06,
+                                   0x04, 0x04, 0x03, 0x04, 0x02, 0x00};
+static u8 anim_throw_bone[] = {0x05, 0x07, 0x06, 0x08, 0x05, 0x09, 0x05,
+                               0x0A, 0x05, 0x0B, 0x05, 0x0C, 0xFF, 0x00};
+static u8 anim_jump1[] = {0x01, 0x01, 0x04, 0x0D, 0x04, 0x0E, 0x01, 0x01, 0x00};
+static u8 anim_jump2[] = {
+    0x01, 0x01, 0x04, 0x0D, 0x06, 0x0E, 0x04, 0x0D, 0x01, 0x01, 0x00};
+static u16 anim_bone_rot[] = {
+    0x0100, 0x0080, 0x0048, 0x0020, 0x0040, 0x0010, 0x0030, 0x0000};
+static s8 dead_parts_selector[] = {
+    0x30, 0x20, 0x14, 0x0C, 0x18, 0x10, 0x20, 0x00};
+static s32 dead_parts_velocity_x[] = {
+    FIX(.75), FIX(1.75), FIX(1.5), FIX(1), FIX(2), FIX(1.75), 0};
+static s32 dead_parts_velocity_y[] = {
+    FIX(-5), FIX(-3), FIX(-2), FIX(-3), FIX(-4), FIX(-.875), FIX(-4)};
+static u16 dead_parts_pos_x[] = {-4, 0, 4, -4, -4, 4, 0, 0};
+static u16 dead_parts_pos_y[] = {-16, -8, -4, -4, 9, 9, 0, 0};
+static u8 attack_timer_cycles[2][4] = {
+    {0x80, 0x08, 0x08, 0x40}, {0xF0, 0xC0, 0xA0, 0x80}};
+static s16 sensors_ground[][2] = {{0, 19}, {8, 0}};
+static s16 sensors_move[][2] = {{-12, 16}, {0, -16}, {0, -16}};
+
+void SlingerAttackCheck(Entity* self) {
+    s32 groundCollision = UnkCollisionFunc2(sensors_ground);
+    s16 moveCollision = UnkCollisionFunc(sensors_move, 3);
+
+    if ((groundCollision & 0xFFFF) == 128 && (moveCollision & 2) != 0) {
+        SetStep(SLINGER_JUMP);
+    } else if (--self->ext.skeleton.attackTimer == 0) {
+        SetStep(SLINGER_ATTACK);
+    }
+}
+
+extern u16 g_EInitSlinger[];
+void EntitySlinger(Entity* self) {
+    Entity* newEntity;
+    u8 animStatus;
+    u8 i;
+
+    if (self->flags & FLAG_DEAD) {
+        self->step = SLINGER_DESTROY;
+    }
+
+    switch (self->step) {
+    case SLINGER_INIT:
+        InitializeEntity(g_EInitSlinger);
+        self->ext.skeleton.attackTimer = 80; // Slinger attack timer cycle
```

Comment:
More of a note than anything, but this extension is used now across skeleton, bone scimitar and now slinger.
Since it was already used in bone scimitar and skeleton I decided to also use it here, but I can create a new extension if preferred.